http_server.py

```diff
@@ -8,6 +12,33 @@
 server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
 
 
+def resolve_uri(uri):
+    uri = os.path.join(ROOT, uri)
```

Comment:
Is this working for you?  I'm surprised if it is.  I expect that with a real browser, the `uri` is always going to be an absolute path and this is what I get with an absolute path uri:

```python
>>> import os.path
>>> os.path.join('./webroot', '/foo/bar/baz')
'/foo/bar/baz'
```